platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+     * <p>
+     * This SHALL be -1 if birth round mode has not yet been enabled.
+     */
+    int64 lowest_judge_generation_before_birth_round_mode = 15;
+}
+
+
+/**
+ * A consensus node software version.
+ *
+ * This message records version information for configuration, the Hedera
+ * API, and the "Services" subsystems.
+ */
+message SoftwareVersion {
+    /**
+     * A single numeric version.<br/>
+     * This is the current configuration version read on node startup.
+     */
+    uint32 config_version = 1;
+
+    /**
+     * A semantic version entry.<br/>
+     * This is the version of the HAPI module (Hedera API).
+     */
+    proto.SemanticVersion hapi_version = 2;
+
+    /**
+     * A semantic version entry.<br/>
+     * This is the version of the services module.
+     */
+    proto.SemanticVersion services_version = 3;
+}
+
+
+/**
+ * A consensus snapshot.<br/>
+ * This is a snapshot of the consensus state for a particular round.
+ *
+ * This message SHALL record consensus data necessary for restart
+ * and reconnect.
+ */
+message ConsensusSnapshot {
+    /**
+     * A consensus round.<br/>
+     * The round number of this snapshot.
+     */
+    uint64 round = 1;
+
+    /**
+     * A list of SHA-384 hash values.<br/>
+     * The hashes of all judges for this round.
+     * <p>
+     * This list SHALL be ordered by creator ID.<br/>
+     * This list MUST be deterministically ordered.
+     */
+    repeated bytes judge_hashes = 2;
+
+    /**
+     * A list of minimum judge information entries.<br/>
+     * These are "minimum ancient" entries for non-ancient rounds.
+     */
+    repeated MinimumJudgeInfo minimum_judge_info_list = 3;
+
+    /**
+     * A single consensus number.<br/>
+     * The consensus order of the next event to reach consensus.
+     */
+    uint64 next_consensus_number = 4;
+
+    /**
+     * A "consensus" timestamp.<br/>
+     * The consensus timestamp of this snapshot.
+     * <p>
+     * This SHALL be a consensus value and MAY NOT correspond to an actual
+     * "wall clock" timestamp.<br/>
+     * Consensus Timestamps SHALL always increase.
+     */
+    proto.Timestamp consensus_timestamp = 5;
+}
+
+/**
+ * Minimum ancient threshold information for round judges.
+ * This message records the minimum ancient threshold agreed by all judges in
+ * a round.
+ */
+message MinimumJudgeInfo {
+    /**
+     * A consensus round.<br/>
+     * The round this judge information applies to.
+     */
+    uint64 round = 1;
+
+    /**
+     * Minimum ancient threshold for all judges in the round.
+     * <p>
+     * This SHALL reflect the relevant minimum threshold, whether
+     * generation-based or birth-round-based.
+     */
+    uint64 minimum_judge_ancient_threshold = 2;
+}
+
+/**
+ * A network address book.<br/>
+ * The address book records the address of every known consensus node that
+ * participates in the network, including `0 weight` nodes.<br/>
+ */
+message AddressBook {
```

Comment:
The plan has been to the name of this object to `Roster` in the platform (in order to distinguish between this and Hedera's concept of an address book). Since protobufs are hard to change, it might make sense to change the name now.

Reply:
agreed.  I'd like to discuss this in a call.   The following things have roster implications: 
1. DAB Phase 2
2. DAB Phase 3
3. TSS  

Relevant changes from DAB Phase 2: 
* Internal/External hostname are replaced with an array of up to 10 ServiceEndpoint protobuf objects. 

Here is a PR that iterates on the protobuf definitions in DAB Phase 2.    We should probably re-use any primitives that the app is using for protobuf representation. 

https://github.com/hashgraph/hedera-protobufs/pull/346/files

Reply:
I renamed this message to `Roster`.